tests/rifCollectorAppTest.php

```diff
@@ -0,0 +1,45 @@
+declare(strict_types=1);
+
+<?php
+
+require_once 'rif-collector-app.php';
+
+use PHPUnit\Framework\TestCase;
+
+class rifCollectorAppTest extends TestCase {
+
+    public function testPopulateTableSuccess(): void
+    {
+        $inputArray = [['make' => 'MAKE',
+            'model'=> 'MODEL',
+            'type'=> 'TYPE',
+            'color' => 'COLOR',
+            'mags_owned' => 2,
+            'power_source' => 'POWER SOURCE',
+            'sites_visited' => 1,
+            'purchase_date' => '2024-01-01']];
+
+        $expected = "<tr>".
+            "<td class='entry make-entry'>MAKE</td>".
+            "<td class='entry model-entry'>MODEL</td>".
+            "<td class='entry type-entry'>TYPE</td>".
+            "<td class='entry color-entry'>COLOR</td>".
+            "<td class='entry mags-entry'>2</td>".
+            "<td class='entry power-entry'>POWER SOURCE</td>".
+            "<td class='entry sites-entry'>1</td>".
+            "<td class='entry purchase-entry'>2024-01-01</td>".
+            "</tr>";
+
+        $actual = populateTable($inputArray);
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testPopulateTableMalformed(): void
+    {
```

Comment:
what about testing the wrong kind of array? What happens in your malformed test if you pass an indexed array as opposed to an assoc array?

Reply:
Believe in that malformed test you described the indexed array would still pass as it's still an array.